src/js/stores/AppFormStore.js

```diff
@@ -186,6 +186,36 @@ const resolveAppKeyToFieldIdMap = {
   "healthChecks": "healthChecks"
 };
 
+// Validate all fields in form store and update validationErrorIndices.
+// Return true if all checks passed, otherwise false.
+function checkAllFieldsForValidity(fields) {
+  return Object.keys(fields).reduce((memo, fieldId) => {
```

Comment:
How about rewriting this to use `.every`? That would stop at the first falsy value, saving some cycles? https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/every